lib/streamlit/legacy_caching/hashing.py

```diff
@@ -628,11 +629,14 @@ def _to_bytes(self, obj: Any, context: Optional[Context]) -> bytes:
 
             h = hashlib.new("md5")
 
-            if self._file_should_be_hashed(obj.__code__.co_filename):
+            code = getattr(obj, "__code__", None)
+            assert code is not None
```

Comment:
I don't think this assert is needed, but it honestly is the truth, especially because we grab something from underneath